### ISSUE TITLE
Tik Tok failed to parse JSON in US

### ISSUE THREAD

**Issue by darktrend** [yt-dlp/yt-dlp#2135]
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2021.12.27**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

USA

### Description

Tiktok having issues, in the US, using VPN or not it throws the same error.

### Verbose log

```shell
yt-dlp --verbose https://www.tiktok.com/@therock
[debug] Command-line config: ['--verbose', 'https://www.tiktok.com/@therock']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, err utf-8, pref cp1252
[debug] yt-dlp version 2021.12.27 [6223f67] (win_exe)
[debug] Python version 3.8.10 (CPython 64bit) - Windows-10-10.0.19042-SP0
[debug] exe versions: ffmpeg 4.2.3
[debug] Optional libraries: Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [tiktok:user] Extracting URL: https://www.tiktok.com/@therock
[tiktok:user] therock: Downloading webpage
[download] Downloading playlist: therock
[tiktok:user] therock: Downloading user video list page 1
WARNING: [tiktok:user] Expecting value: line 1 column 1 (char 0). Retrying...
[tiktok:user] therock: Downloading user video list page 1 (attempt 1)
WARNING: [tiktok:user] Expecting value: line 1 column 1 (char 0). Retrying...
[tiktok:user] therock: Downloading user video list page 1 (attempt 2)
WARNING: [tiktok:user] Expecting value: line 1 column 1 (char 0). Retrying...
[tiktok:user] therock: Downloading user video list page 1 (attempt 3)
ERROR: therock: Failed to parse JSON  (caused by JSONDecodeError('Expecting value: line 1 column 1 (char 0)')); please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; type  yt-dlp -U  to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "yt_dlp\extractor\common.py", line 990, in _parse_json
  File "json\__init__.py", line 357, in loads
  File "json\decoder.py", line 337, in decode
  File "json\decoder.py", line 355, in raw_decode
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
Traceback (most recent call last):
  File "yt_dlp\extractor\common.py", line 990, in _parse_json
  File "json\__init__.py", line 357, in loads
  File "json\decoder.py", line 337, in decode
  File "json\decoder.py", line 355, in raw_decode
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "yt_dlp\YoutubeDL.py", line 1353, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1640, in <lambda>
  File "yt_dlp\utils.py", line 2620, in __getitem__
  File "yt_dlp\extractor\tiktok.py", line 438, in _entries_api
  File "yt_dlp\extractor\tiktok.py", line 75, in _call_api
  File "yt_dlp\extractor\common.py", line 979, in _download_json
  File "yt_dlp\extractor\common.py", line 965, in _download_json_handle
  File "yt_dlp\extractor\common.py", line 994, in _parse_json
yt_dlp.utils.ExtractorError: therock: Failed to parse JSON  (caused by JSONDecodeError('Expecting value: line 1 column 1 (char 0)')); please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; type  yt-dlp -U  to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
```

**Comment by rajatvermaiam** [comment]
i think tiktok changed the way they give response of aweme API.

**Comment by aidenmitchell** [comment]
Same issue in Canada.

**Comment by matthewladams** [comment]
I'm not sure if the API changes just happen to correlate with it, but the fallback to the webpage failing seems to be due to a change in framework that's being rolled out from NextJS to Sigi. I haven't got this working fully yet but seems to work for individual videos https://github.com/matthewladams/yt-dlp/commit/c494f5736dc69d6bff9f03720e6430a7d523c3ec

**Comment by SelfAdjointOperator** [comment]
Same in the UK

**Comment by DananaBanana** [comment]
I'm currently using [this](https://github.com/ytdl-patched/yt-dlp/releases/tag/2021.12.29.114514) release of ytdlp-patched which includes #2142, I can confirm that this fix works on some videos but not all.
I personally don't know why this one video works and the others don't.
This is one that does work: https://www.tiktok.com/@okdonelij/video/7046409408393481518
Here are some examples of the ones that don't: 
- https://www.tiktok.com/@cartimindset/video/7045845983581867310
- https://www.tiktok.com/@killacyst/video/7046376215359343878
- https://www.tiktok.com/@doberman23456/video/7045593175196486917

For these videos I got the same error as first listed (after a quick glance)
Here is the full verbose log of one of the videos that didn't work:
```
[debug] Command-line config: ['https://www.tiktok.com/@cartimindset/video/7045845983581867310', '--verbose']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, err utf-8, pref cp1252
[debug] yt-dlp version 2021.12.01 [91f071a] (win_exe)
[debug] Python version 3.8.10 (CPython 64bit) - Windows-10-10.0.22000-SP0
[debug] exe versions: ffmpeg 4.3, ffprobe 4.3
[debug] Optional libraries: Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [TikTok] Extracting URL: https://www.tiktok.com/@cartimindset/video/7045845983581867310
[TikTok] 7045845983581867310: Downloading video details
WARNING: [TikTok] 7045845983581867310: Failed to parse JSON  (caused by JSONDecodeError('Expecting value: line 1 column 1 (char 0)')); please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; type  yt-dlp -U  to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.; Retrying with webpage
[TikTok] 7045845983581867310: Downloading webpage
[TikTok] 7045845983581867310: Downloading video webpage
ERROR: [TikTok] 7045845983581867310: Unable to extract json_string; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; type  yt-dlp -U  to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; type  yt-dlp -U  to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "yt_dlp\extractor\common.py", line 597, in extract
  File "yt_dlp\extractor\tiktok.py", line 357, in _real_extract
  File "yt_dlp\extractor\common.py", line 1171, in _search_regex
yt_dlp.utils.RegexNotFoundError: Unable to extract json_string; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; type  yt-dlp -U  to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "yt_dlp\extractor\common.py", line 597, in extract
  File "yt_dlp\extractor\tiktok.py", line 357, in _real_extract
  File "yt_dlp\extractor\common.py", line 1171, in _search_regex
yt_dlp.utils.RegexNotFoundError: Unable to extract json_string; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; type  yt-dlp -U  to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "yt_dlp\YoutubeDL.py", line 1329, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1398, in __extract_info
  File "yt_dlp\extractor\common.py", line 623, in extract
yt_dlp.utils.RegexNotFoundError: [TikTok] 7045845983581867310: Unable to extract json_string; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; type  yt-dlp -U  to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; type  yt-dlp -U  to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
```

Hopefully someone can use this.

**Comment by pukkandan** [comment]
@MinePlayersPE @llacb47 While this is fixed via the API, it might be best to look into how to fix this also in the webpage extraction. According to https://github.com/ytdl-org/youtube-dl/issues/30251, the website has been experimenting with Sigi for a while now
